### PR TITLE
Use custom `mount.sh` script instead of `pmount`

### PR DESCRIPTION
**Pull request by adghayes**
We are currently using `pmount` and `pumount` to mount and unmount USB drives, respectively. They have some limitations:

1. We cannot mount exFAT drives
2. We cannot mount loop devices, which we may want to do during development

The first advantage of `pmount` was that it does not need to be run as the root, like `mount`. We can solve that by giving the right users `sudo` privileges for `mount`, which will happen in `vxsuite-complete-system` and our development environments.

The second advantage of `pmount` is that you don't need to create or even specify a mount point, it will do that for you. My fix is to have a single mount point that we use for all USB drives. Currently we use mount points like `/media/usb-drive-sda1` which don't communicate meaningful information. In this PR, we start using `/media/vx/usb-drive` as a mount point. It's better to use a subdirectory of `/media`, like `/media/vx`, so that we can give permissions to that subdirectory and in the future maybe create the mount points on the fly. But for simplicity, for now, just a constant mount point because that meets our requirements.

### CODE REVIEW

File: src/ipc/mount-usb-drive.ts [hunk 1]
```diff
@@ -9,13 +9,18 @@ export interface Options {
   label?: string;
 }
 
+const MOUNT_DIRECTORY = '/media/vx/usb-drive';
```
Comment (adghayes) [inline src/ipc/mount-usb-drive.ts:12]: This is a constant which exists across all three repos...

File: src/ipc/mount-usb-drive.ts [hunk 2]
```diff
@@ -9,13 +9,18 @@ export interface Options {
...
+const MOUNT_OPTIONS = ['umask=000', 'nosuid', 'nodev', 'noexec'];
```
Comment (adghayes) [inline src/ipc/mount-usb-drive.ts:13]: These options are protections mimicking the same way we've been mounting with `pmount`.

File: README.md [hunk 3]
```diff
@@ -34,6 +34,10 @@ Presents a file save dialog to the user and, if a file is chosen, resolves to an
 
 Gets a list of USB drives and, if mounted, where. To mount or unmount a drive, pass its device name to `kiosk.mountUsbDrive` or `kiosk.unmountUsbDrive`.
 
+## `sudo` Permissions
```
Comment (carolinemodic) [inline README.md:37]: thanks for adding

File: src/ipc/mount-usb-drive.ts (hunk 1 above)
Comment (carolinemodic) [reply src/ipc/mount-usb-drive.ts:12]: oof

**Review by carolinemodic — APPROVED** [review]
Would be curious to get @benadida and/or @amcmanus thoughts on how to handle the fact that we need to specify a mount point. But your solution is the only thing I can really think of to do.

**Comment by adghayes** [comment]
@carolinemodic we don't necessarily need to specify a mount point ahead of time. We could make it on the fly! In fact, what I have of for permissions (similar in development and complete system) are:

```bash
# Create mount point for kiosk-browser to mount USB drives
sudo mkdir -p /media/vx/usb-drive
sudo chown -R vx-ui:vx-group /media/vx
```
We could instead just create `/media/vx` and then `kiosk-browser` could create subdirectories within `media/vx` and mount in those subdirectories, and remove after mounting. That just added a complexity that I didn't see benefit for at the moment, but if we ever, say mount two USB drives, we'll have to do that. 

I'll wait to merge this to see if @benadida or @eventualbuddha have objections to the current approach.

**Review by benadida — APPROVED** [review]
I'm a little bit worried about giving `sudo` to `kiosk-browser` for `mount`, which is a very powerful command. Could lead to mounting different partitions as `/`, for example, or other nasty attacks.

I'd like to see us put the specific mounting command into a shell script, and us give sudo to that shell script, not to all of `mount`.

I'm comfortable doing this as a followup ticket, but let's tag it security and prioritize it.

**Comment by adghayes** [comment]
@benadida thanks for the suggestion. I'd like to do that as part of this instead of a follow-up ticket.

This is simple enough to do from `vxsuite-complete-system`. But I'm wondering what the cleanest way to handle this from our dev environments is. Two ideas:
- Have a mounting and unmounting script in `vxsuite` or `kiosk-browser` that we pass in as an argument to `kiosk-browser`
- Use a signing script when passed, just use `sudo mount` when none is available like in our dev environments

cc @amcmanus

**Comment by benadida** [comment]
> @benadida thanks for the suggestion. I'd like to do that as part of this instead of a follow-up ticket.
> 
> This is simple enough to do from `vxsuite-complete-system`. But I'm wondering what the cleanest way to handle this from our dev environments is. Two ideas:
> 
>     * Have a mounting and unmounting script in `vxsuite` or `kiosk-browser` that we pass in as an argument to `kiosk-browser`
> 
>     * Use a signing script when passed, just use `sudo mount` when none is available like in our dev environments
> 
> 
> cc @amcmanus

I like idea #1. Matches other stuff we do.

**Comment by adghayes** [comment]
@benadida can you point to an example?

It looks like we do something similar for our signing script, but I don't see anywhere in `vxsuite-complete-system` or otherwise we are actually passing in our signing script to `kiosk-browser`?

**Comment by benadida** [comment]
> @benadida can you point to an example?
> 
> It looks like we do something similar for our signing script, but I don't see anywhere in `vxsuite-complete-system` or otherwise we are actually passing in our signing script to `kiosk-browser`?

Yeah we ripped that out cause it's in transition. But we had it at some point I believe. Ultimately, the idea of passing a script path to kiosk browser is something i want to use for signing too, so i like that direction.

**Comment by adghayes** [comment]
@benadida one last thought, in the spirit of challenging your security recommendations. 

The situation you describe, of mounting a partition as `/`, is not actually possible with this `kiosk-browser` API. The `kiosk-browser` API takes the device to mount, but does not allow specification of the mounting directory (which will always be `/media/vx/[something]`. 

However in `vx-complete-system`, `vx-ui` would have unfettered access to mount. So if someone managed to open up a shell, that situation is possible.

Does that affect your thinking at all?

**Comment by benadida** [comment]
> @benadida one last thought, in the spirit of challenging your security recommendations.
> 
> The situation you describe, of mounting a partition as `/`, is not actually possible with this `kiosk-browser` API. The `kiosk-browser` API takes the device to mount, but does not allow specifying the mounting directory (which will always be `/media/vx/[something]`.
> 
> However in `vx-complete-system`, `vx-ui` would have unfettered access to mount. So if someone managed to open up a shell, that situation is possible.

Thanks for pushing on this!

What I'm worried about is exactly the last bit you write. I want the `vx-ui` user to have limited permissions, so that if someone gets shell access, they can't go on a rampage.

Thinking about this some more – didn't we say we wanted to move all of this, including USB mounting/unmounting, to backend services, and no longer `kiosk-browser` APIs? I think that can be done as a followup task, but just making sure we remember to do that.

**Comment by adghayes** [comment]
Thanks for clarifying!

Yeah, we want to move it to the backend services. We're not quite ready to do that yet (we need to stand up `grout` backends for all the backends first, even if we don't fully migrate). I think we will move over all USB functionality and possibly other functionality together in a follow-up.

Related to the formatting work, do you take issue with any of these `sudoer` permissions in `vxsuite-complete-system`?

```
vx-ui ALL=(root:ALL) NOPASSWD: /usr/sbin/sfdisk
vx-ui ALL=(root:ALL) NOPASSWD: /usr/sbin/mkfs.fat
vx-ui ALL=(root:ALL) NOPASSWD: /usr/sbin/mkfs.exfat
```

Formatting could also be moved to a script in a follow-up.

**Comment by amcmanus** [comment]
> > @benadida thanks for the suggestion. I'd like to do that as part of this instead of a follow-up ticket.
> > This is simple enough to do from `vxsuite-complete-system`. But I'm wondering what the cleanest way to handle this from our dev environments is. Two ideas:
> > ```
> > * Have a mounting and unmounting script in `vxsuite` or `kiosk-browser` that we pass in as an argument to `kiosk-browser`
> > 
> > * Use a signing script when passed, just use `sudo mount` when none is available like in our dev environments
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > cc @amcmanus
> 
> I like idea #1. Matches other stuff we do.

Agreed. We should keep `sudo` as narrowly scoped as possible. As long as we confirm permissions on the scripts are appropriate, it's a much more manageable approach than blanket sudo for commands.

**Comment by amcmanus** [comment]
> @carolinemodic we don't necessarily need to specify a mount point ahead of time. We could make it on the fly! In fact, what I have for permissions (similar in development and complete system) is currently:
> 
> ```shell
> # Create mount point for kiosk-browser to mount USB drives
> sudo mkdir -p /media/vx/usb-drive
> sudo chown -R vx-ui:vx-group /media/vx
> ```
> 
> We could instead _only_ create `/media/vx` instead of `/media/vx/usb-drive` and then `kiosk-browser` could create subdirectories within `/media/vx` and mount in those subdirectories, and remove after mounting. That just added a complexity that I didn't see benefit for at the moment, but if we ever, say mount two USB drives, we'll have to do that.
> 
> I'll wait to merge this to see if @benadida, @amcmanus, or @eventualbuddha have objections to the current approach.

No objection to this approach, but in the longer term, I think we should consider other options that minimize or eliminate the need to maintain constants across multiple repos. That's a problem for later though, no need to block this.

**Comment by adghayes** [comment]
@benadida @carolinemodic this PR has changed substantially, re-requesting review.

File: src/ipc/mount-usb-drive.ts (hunk 1 above)
Comment (adghayes) [reply src/ipc/mount-usb-drive.ts]: Now, with the script changes that @benadida suggested, this constant only exists in complete system

**Review by benadida — APPROVED** [review]

**Pull request merged**